tests/unit/s2n_mem_usage_test.c

```diff
@@ -254,21 +254,25 @@ int main(int argc, char **argv)
     free(clients);
     free(servers);
 
-    TEST_DEBUG_PRINT("\n");
```

Comment:
I think we want TEST_DEBUG_PRINT here. Without this, all of this output will print with every test run, which isn't great for normal users running tests. If this information is difficult to determine after running this in CI, could we instead update the CI to set `S2N_TEST_DEBUG`?

Reply:
`TEST_DEBUG_PRINT` [doesn't work](https://github.com/aws/s2n-tls/actions/runs/7590436165/job/20679054682#step:3:2541) with CMake, which is how our unit tests are run. CMake will capture `stdout` by default and then output it only if a test fails (when using the `--output-on-failure` option).